File: src/test/java/randomidprovider/RandomParameterReaderTest.java
package randomidprovider;

import org.junit.Test;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;
import static org.junit.Assert.fail;

/**
 *
 */
public class RandomParameterReaderTest {

    private File generatePlainFile() {

        String tempDir = System.getProperty("java.io.tmpdir");
        System.out.println(tempDir);
        File f = new File(tempDir, "names.txt");
        f.deleteOnExit();
        try {
            BufferedWriter bos = new BufferedWriter(new FileWriter(f));
<<<<<<< HEAD
            bos.write("Jan;0.25\nPiet;0.5\nKlaas;0.25");
=======
            for (int i = 0;i<100;i++) {
                bos.write("Naam" + i);
                bos.newLine();
            }
          bos.flush();
>>>>>>> c507a8010582a4fa503d396e4ee2d6d54b704ae2
            bos.close();
        } catch (IOException e) {
            e.printStackTrace();  //To change body of catch statement use File | Settings | File Templates.
        }

        return f;
    }

    @Test
    public void GenerateNames() {
        try {
            File f = generatePlainFile();
            RandomParameterReader rpr = new RandomParameterReader(f, new Random(0));
<<<<<<< HEAD
            for (int i = 0; i < 20; i++) {
                System.out.println(rpr.getOne());
            }
=======
            for (int i =0;i<19;i++) {
               rpr.getOne();
            }

            assertThat(rpr.getOne(), is("Naam77"));
>>>>>>> c507a8010582a4fa503d396e4ee2d6d54b704ae2
        } catch (IOException e) {
            fail();
        }
    }

}

File: Generator/src/test/scala/RandomIdProvider/Entities/RandomParameterReaderTest.scala
package RandomIdProvider.Entities

import junit.framework.{TestCase, TestSuite, Test}
import junit.framework.Assert._
import java.io._
import util.Random
import io.{BufferedSource, Codec, Source}
import scalax.io._
import RandomIdProvider.generator.RandomParameterReader

/**
 *
 */
object RandomParameterReaderTest {

  def suite: Test = {
    val suite = new TestSuite(classOf[RandomDateGeneratorTest]);
    suite
  }

  def main(args: Array[String]) {
    junit.textui.TestRunner.run(suite);
  }
}

/**
 * Unit test for simple App.
 */
class RandomParameterReaderTest extends TestCase("app") {

  /**
   * Rigourous Tests :-)
   */

  def generatePlainFile(): File = {
    val tempDir = System.getProperty("java.io.tmpdir");

    val f = new File(tempDir, "names.txt") :File
    f.deleteOnExit();

    val out = new PrintWriter(f)
    out.write("Jan;0.25\nPiet;0.5\nKlaas;0.25")
    out.close()

    f;
  }

  def generatePlainFileWithoutProbs: File = {
    val tempDir = System.getProperty("java.io.tmpdir");

    val f = new File(tempDir, "names.txt") :File
    f.deleteOnExit();

    val out = new PrintWriter(f)
    out.write("Jan\nPiet\nKlaas")
    out.close()

    f;
  }

  def testGenerateNames() {
      val f = generatePlainFile()
      val s =  Source.fromFile(f)
      val rpr = new RandomParameterReader(s, new Random(0));
      assertEquals(rpr.getOne,"Piet")
      assertEquals(rpr.getOne,"Jan")


  }

  def testGenerateNamesWithoutProbs() {
    val f = generatePlainFileWithoutProbs
    val s =  Source.fromFile(f)
    val rpr = new RandomParameterReader(s, new Random(0));
    assertEquals(rpr.getOne,"Klaas")
    assertEquals(rpr.getOne,"Jan")


  }
}
